lib/credentials/shared_ini_file_credentials.js

```diff
@@ -175,6 +180,33 @@ AWS.SharedIniFileCredentials = AWS.util.inherit(AWS.Credentials, {
     }
   },
 
+  /**
+  * Executes the credential_process and retrieves
+  * credentials from the output
+  * @api private
+  * @param profile [map] credentials profile
+  * @throws SharedIniFileCredentialsProviderFailure
+  */
+  loadViaCredentialProcess: function loadCredentialProcess(profile, callback) {
+    try {
+      var output = proc.execSync(profile['credential_process']);
+      var credData = JSON.parse(output);
+      if (parseInt(credData.Version) !== 1) {
+        throw AWS.util.error(
+          Error('credential_process does not return Version == 1 for profile ' + this.profile),
+          { code: 'SharedIniFileCredentialsProviderFailure'}
+        );
+      } else {
+        this.accessKeyId = credData.AccessKeyId;
+        this.secretAccessKey = credData.SecretAccessKey;
+        this.sessionToken = credData.SessionToken;
```

Comment:
Is `Expiration` handled elsewhere?

I expected to see something like the following here:
```js
if (credData.Expiration) {
  this.expireTime = new Date(credData.Expiration);
}
```

https://docs.aws.amazon.com/cli/latest/topic/config-vars.html#sourcing-credentials-from-external-processes

Reply:
Yep, I'll be updating this PR with additional tests and functionality to cover the expireTime.